pkg/beholder/client.go

```diff
@@ -0,0 +1,299 @@
+package beholder
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	otellog "go.opentelemetry.io/otel/log"
+	otelmetric "go.opentelemetry.io/otel/metric"
+	sdklog "go.opentelemetry.io/otel/sdk/log"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	sdkresource "go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	oteltrace "go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type Emitter interface {
+	// Sends message with bytes and attributes to OTel Collector
+	Emit(ctx context.Context, body []byte, attrKVs ...any) error
+}
+
+type messageEmitter struct {
+	messageLogger otellog.Logger
+}
+
+type Client struct {
+	Config Config
+	// Logger
+	Logger otellog.Logger
+	// Tracer
+	Tracer oteltrace.Tracer
+	// Meter
+	Meter otelmetric.Meter
+	// Message Emitter
+	Emitter Emitter
+
+	// Providers
+	LoggerProvider        otellog.LoggerProvider
+	TracerProvider        oteltrace.TracerProvider
+	MeterProvider         otelmetric.MeterProvider
+	MessageLoggerProvider otellog.LoggerProvider
+
+	// OnClose
+	OnClose func() error
+}
+
+// NewClient creates a new Client with initialized OpenTelemetry components
+// To handle OpenTelemetry errors use SetOtelErrorHandler
```

Comment:
I think it would be OK to just link to the otel func here and drop the `SetOtelErrorHandler` helper, in order to keep this package API simpler.
```suggestion
// To handle OpenTelemetry errors use [otel.SetErrorHandler]
```